### DISCUSSION TITLE
When dpgen is running, why is there no virial file in the file converted from dpdata?

### DISCUSSION THREAD

**Discussion by 12jscvb** [deepmodeling/dpdata#600] [Q&A]
After calculating the single-point energy structure at the end of each cycle, dpdata converts it into npy format files, but only four files, box, energy,coord and force, and no virial file, while dpgen sets stress_tensor in cp2k input file. Besides, when I manually transfer the ouput file, virial.npy will appear. What is the reason for this?  Thanks!  Best wishes

**Comment by robinzyb** [comment]
You may not print stresses, although cp2k calculated the stresses.
https://manual.cp2k.org/trunk/CP2K_INPUT/FORCE_EVAL/PRINT/STRESS_TENSOR.html

**Reply by 12jscvb** [reply]
This is a cp2k input file automatically generated by dpgen, with the setting stress_tensor in the print module. But dpgen uses dpdata to convert the output file without viriral.npy. However, I manually convert the output file to produce virial.npy. thank you for reply
[input.txt](https://github.com/deepmodeling/dpdata/files/14024094/input.txt)

**Reply by robinzyb** [reply]
Could you paste the code for manually converting cp2k to dpdata with virial.npy? Also have you installed cp2kdata

**Reply by 12jscvb** [reply]
yes,I have installed the cp2kdata,  the first file is converting cp2k to dpdata,   the second file is output file 
[deal.txt](https://github.com/deepmodeling/dpdata/files/14024766/deal.txt)
[output.txt](https://github.com/deepmodeling/dpdata/files/14024774/output.txt)

**Reply by robinzyb** [reply]
the code works properly, this is exactly the same code used in dpgen.
```python
import dpdata
#cp2kmd_dir = "./"
#cp2kmd_output_name = "new.out"
dp = dpdata.LabeledSystem("./output.txt",fmt="cp2kdata/e_f")
dp.to_deepmd_npy('test',fmt='deepmd-npy')
#print(cp2koutput)
print(dp)
```
```shell
--- You are parsing data using package Cp2kData ---
Atom names are fake chemical symbols as you set in cp2k input.
--- You are parsing data using package Cp2kData ---
Data Summary
Labeled System
-------------------
Frame Numbers      : 1
Atom Numbers       : 255
Including Virials  : Yes
Element List       :
-------------------
Al  N  H  Cl  O
55  20  80  20  80

```
have no idea where goes wrong. Could you try to make a new folder, for example ` test_dpgen`

and copy the files of iteration.00000 to the new folder, as well as some dpgen inputs.
Then modify the `dpgen.record` to 

```
0 0
0 1
0 2
0 3
0 4
0 5
0 6
0 7
```

and restart the dpgen. 
See if the dpgen can correctly genreate `virial.npy`.

**Reply by 12jscvb** [reply]
Ok, this is also the case when I used dpgen in the Bohrium platform before. The dpgen processing file does not contain virial. In addition, the initial data set contains virial information while the dpgen iteration process does not contain virial information.  Does this affect the model?

**Reply by robinzyb** [reply]
It depends whether the npt simulations are used. If prediction of pressure is necessary, the answer is yes. If not, just ignore it. You wouldn't use it in the deepmd fitting.

**Reply by 12jscvb** [reply]
thank Professor for  reply.  Benefit a lot

**Reply by robinzyb** [reply]
Has your problem been solved? Does the `virial.npy` file appear in your new DP-GEN run?

**Comment by 12jscvb** [comment]
Unfortunately, Virial.npy still does not appear in new DP-GEN run&nbsp;



---Original---
From: "Yongbin ***@***.***&gt;
Date: Tue, Feb 6, 2024 06:25 AM
To: ***@***.***&gt;;
Cc: "Express ***@***.******@***.***&gt;;
Subject: Re: [deepmodeling/dpdata] When dpgen is running, why is there novirial file in the file converted from dpdata? (Discussion #600)




 
Has your problem been solved? Does the virial.npy file appear in your new DP-GEN run?

**Reply by robinzyb** [reply]
It is unusual. 
Are the dpgen and dpdata and cp2kdata installed in the same environment? If so, it should be ok.

**Comment by 12jscvb** [comment]
I am using cp2k2023 version, will it be relevant. In addition, I found no cp2kdata installation package by pip list command, but dpgen automatically converted data without error (dpgen related packages were installed by pip install deepmd-kit command)




---Original---
From: "Yongbin ***@***.***&gt;
Date: Wed, Feb 21, 2024 15:49 PM
To: ***@***.***&gt;;
Cc: "Express ***@***.******@***.***&gt;;
Subject: Re: [deepmodeling/dpdata] When dpgen is running, why is there novirial file in the file converted from dpdata? (Discussion #600)




 
It is unusual.
 Are the dpgen and dpdata and cp2kdata installed in the same environment? If so, it should be ok.

**Reply by robinzyb** [reply]
try to install cp2kdata, could fix this problem. (dpdata may not recongnize the updated cp2k version

**Reply by 12jscvb** [reply]
ok, thank you for help. I will try it
 Could I ask you one more question? In the combustion direction, cp2k runs 2000 steps of aimd at 500k 1000k 1500k 2000k temperature respectively, with step size of 0.1fs, extracts a structure every 100 steps, a total of 84 initial data, and then dpdata converts it into npy data, that is, the initial data set is 84. 
In the dpgen active learning stage, I set up a total of six loops, and found that the accuracy was getting worse and worse(the attachments are param.json file and nohup file). 
May I ask why this is? Is it because the number of my data sets is too small

[nohup.txt](https://github.com/deepmodeling/dpdata/files/14357690/nohup.txt)
[param.txt](https://github.com/deepmodeling/dpdata/files/14357952/param.txt)

**Reply by robinzyb** [reply]
Could you try to start with nvt of low temperature like 200/300/500k and less steps like 1000/2000. and gradually increase nvt tempetures over several iterations. The potentials are too broken to be able to drive such fierce simulations.

**Reply by 12jscvb** [reply]
thank you relpy.  Your meanings is that in the first few loops the temp parameter in the param.json file is set like "200 300 500", and as the potential function is improved, then the temp temperature can be increased, for example, set like "500 700 900". Also, do I need to re-run an aimd of cp2k at 200/300/500k temperature to get the initial data set?

**Reply by robinzyb** [reply]
"do I need to re-run an aimd of cp2k at 200/300/500k temperature to get the initial data set?" not necessary

**Reply by 12jscvb** [reply]
ok,If the initial data set I provide is large enough,such as up to 1,000 or more, could I start with nvt of high temperature like 500/1000/1500k?

**Reply by robinzyb** [reply]
The size of initial data set doesn't matter but the quality of your potetial does

**Reply by 12jscvb** [reply]
Thank you very much for your answer. Benefit a lot
